nixos/modules/services/video/epgstation/default.nix

```diff
@@ -0,0 +1,277 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.epgstation;
+
+  username = config.users.users.epgstation.name;
+  groupname = config.users.users.epgstation.group;
+
+  settingsFmt = pkgs.formats.json {};
+  settingsTemplate = settingsFmt.generate "config.json" cfg.settings;
+  preStartScript = pkgs.writeScript "epgstation-prestart" ''
+    #!${pkgs.runtimeShell}
+
+    PASSWORD="$(head -n1 "${cfg.basicAuth.passwordFile}")"
+    DB_PASSWORD="$(head -n1 "${cfg.database.passwordFile}")"
+
+    # setup configuration
+    touch /etc/epgstation/config.json
+    chmod 640 /etc/epgstation/config.json
+    sed \
+      -e "s,@password@,$PASSWORD,g" \
+      -e "s,@dbPassword@,$DB_PASSWORD,g" \
+      ${settingsTemplate} > /etc/epgstation/config.json
+    chown "${username}:${groupname}" /etc/epgstation/config.json
+
+    # NOTE: Use password authentication, since mysqljs does not yet support auth_socket
+    if [ ! -e /var/lib/epgstation/db-created ]; then
+      ${pkgs.mysql}/bin/mysql -e \
+        "GRANT ALL ON ${cfg.database.name}.* TO ${username}@localhost IDENTIFIED by \"$DB_PASSWORD\";"
```

Comment:
the potential for a misformatted strings to blow up the mysql syntax here is kinda scaring me - could we quote these (I think mysql uses backticks)? and/or do some validation at the option-level to make sure these are only ever valid mysql identifier characters?

Reply:
also, this should potentially use `services.mysql.ensureUsers.*.ensurePermissions`?

Reply:
I've now added quotes. I think a full-blown validation is a overkill because NixOS options have to be trusted by the administrator.

Reply:
`ensurePermissions` is unusable for EPGStation becasue EPGStation doesn't support `auth_socket`.